src/components/UserAccess/UserAccessForm/RoleSection.tsx

```diff
@@ -1,21 +1,30 @@
 import React from 'react';
-import { ExpandableSection, FormSection } from '@patternfly/react-core';
+import { Bullseye, ExpandableSection, FormSection, Spinner } from '@patternfly/react-core';
 import { useField } from 'formik';
 import HelpPopover from '../../../components/HelpPopover';
+import { useRoleMap } from '../../../hooks/useRole';
 import DropdownField from '../../../shared/components/formik-fields/DropdownField';
-import { WorkspaceRole } from '../../../types';
+import { NamespaceRole } from '../../../types';
 import { PermissionsTable } from './PermissionsTable';
-
 import './RoleSection.scss';
 
-const dropdownItems = [
-  { key: 'contributor', value: 'contributor' },
-  { key: 'maintainer', value: 'maintainer' },
-  { key: 'admin', value: 'admin' },
-];
-
 export const RoleSection: React.FC<React.PropsWithChildren<unknown>> = () => {
-  const [{ value: role }] = useField<WorkspaceRole>('role');
+  const [{ value: role }] = useField<NamespaceRole>('role');
+  const [roleMap, roleMapLoading] = useRoleMap();
+
+  if (roleMapLoading) {
+    return (
+      <Bullseye>
+        <Spinner data-test="spinner" />
+      </Bullseye>
+    );
+  }
```

Comment:
again no spinner is needed here, disable dropdown if roleMap is loading with placeholder text, Loading...